### PR TITLE
Allow to use NNCF + deep-object-reid from OTE

### PR DESCRIPTION
**Pull request by LeonidBeynenson**
Key changes:
* add inheritance of YACS config files in OpenMMLab's configs manner
  * add the field `_base_` to config
  * add the function `merge_from_files_with_base`
  * make merging of config files made by `merge_from_files_with_base` everywhere
* make training create symbolic link `model-best.pth.tar` pointing to the best model
  * the best model is the main model with the best Rank-1 accuracy after the last drop of learning rate
  * if mutual learning is used, the symbolic link `model-best.pth.tar` is created in the auxiliary models' folder too
    (but the best epoch is still chosen by the main model's Rank-1 accuracy)
* Shift version of torch and torchvision to
  * torch==1.8.1 
  * torchvision==0.9.1
* Make compression compatible with nncf==1.7.0

### CODE REVIEW

**Comment by LeonidBeynenson** [comment]
@sovrasov , @kprokofi : Vlad, Kirill, please review this PR.

It is required to use NNCF compression of image classification from OTE.

**Review by sovrasov — APPROVED** [review]

File: torchreid/engine/engine.py [hunk 1]
```diff
@@ -348,15 +367,18 @@ def run(
...
 
-                if (self.early_stoping and
-                    not lr_finder and
-                    self.exit_on_plataeu(top1, top5, mAP)):
+                if not lr_finder:
+                    should_exit, is_candidate_for_best = self.exit_on_plateau_and_choose_best(top1, top5, mAP)
+                    should_exit = self.early_stoping and should_exit
+
+                    if self.save_chkpt:
+                        self.save_model(self.epoch, save_dir, is_best=is_candidate_for_best)
```
Comment (kprokofi) [inline torchreid/engine/engine.py:379]: I liked this solution to save the best model based on exit criteria

**Review by kprokofi — APPROVED** [review]
